server/bin/pbench-check-tb-age.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""Pbench Check TB Age
+
+Return 0 if the given tar ball (full path) is within the configured maximum
+age (specified in days), and return 1 if it is not.
+
+A return value > 1 indicates some other error with the command.
+"""
+
+import sys
+import os
+import re
+from datetime import datetime
+from argparse import ArgumentParser
+from configparser import NoOptionError
+
+from pbench import (
+    PbenchConfig,
+    BadConfig,
+)
+
+
+_NAME_ = "pbench-check-tb_age"
+
+tb_pat_r = (
+    r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz"
+)
+tb_pat = re.compile(tb_pat_r)
+
+
+def main(options):
+    if not options.tb_path:
+        print(
+            f"{_NAME_}: ERROR: No tar ball path specified", file=sys.stderr,
+        )
+        return 2
+    tb_path = os.path.realpath(options.tb_path)
+    tb_name = os.path.basename(tb_path)
+
+    if not options.cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable",
+            file=sys.stderr,
+        )
+        return 3
+
+    try:
+        config = PbenchConfig(options.cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 4
+
+    archive = config.ARCHIVE
+    archive_p = os.path.realpath(archive)
+
+    if not archive_p:
+        print(
+            f"The configured ARCHIVE directory, {archive}, does not exist",
+            file=sys.stderr,
+        )
+        return 5
+
+    if not os.path.isdir(archive_p):
+        print(
+            f"The configured ARCHIVE directory, {archive}," " is not a valid directory",
+            file=sys.stderr,
+        )
+        return 6
+
+    incoming = config.INCOMING
+    incoming_p = os.path.realpath(incoming)
```

Comment:
This can be incorporated in one line

Reply:
Are you thinking of using a `;` between statements?

Reply:
incoming = os.path.realpath(config.INCOMING)

Reply:
Isn't `incoming` used in the `print` statement below?